xdsl/dialects/memref_stream.py

```diff
@@ -513,6 +521,12 @@ def print(self, printer: Printer):
                 lambda iterator_type: printer.print_string_literal(iterator_type.data),
             )
             printer.print_string("]")
+            if self.doc:
```

Comment:
We generally prefer `is None` to differentiate between empty cases and `None`, here my recommendation would also be to rather check for this:
```suggestion
            if self.doc.data:
```
And for the default value for the argument to be `StringAttr("")`

Reply:
Having a default value of `StringAttr("")` still prints the attribute:
```
  memref_stream.generic {
    bounds = [4, 2, 3],
    indexing_maps = [
      affine_map<(d0, d1, d2) -> (d0, d2)>,
      affine_map<(d0, d1, d2) -> (d2, d1)>,
      affine_map<(d0, d1) -> (d0, d1)>,
      affine_map<(d0, d1) -> (d0, d1)>
    ],
    iterator_types = ["parallel", "parallel", "reduction"],
    library_call = ""
  } ins(%E, %F : memref<4x2xf64>, memref<2x3xf64>) outs(%G, %I : memref<4x3xf64>, memref<4x3xf64>) inits(%H : f64, None) {
  ^4(%e_1 : f64, %f_1 : f64, %acc_old_1 : f64, %acc_old_2 : f64):
    %prod_1 = arith.mulf %e_1, %f_1 : f64
    %acc_new_1 = arith.addf %acc_old_1, %prod_1 : f64
    %acc_new_2 = arith.addf %acc_old_2, %prod_1 : f64
    linalg.yield %acc_new_1, %acc_new_2 : f64, f64
  }
  ```
As it is an optional attribute, I don't think this is desirable

Reply:
But that's something you added in this PR, you could just not print it if it's the empty string, similarly make the parsing step optional with the same default value

Reply:
The same holds for `--print-op-generic`, where it can't be changed.

Reply:
Ah yes, I wonder what MLIR does there

Reply:
OK it does the same thing, then your approach is best